added user detail page

beanduong opened the pull request:
- added mixing for body titles
- added `hideNumbers` option to hide numbers in `HorizontalEventList`
- added user detail page when calling `/user/:uid`

tbtz commented on src/frontend/pages/UserDetailPage.vue:
```diff
@@ -0,0 +1,55 @@
+    },
+    methods: {
+      async init() {
+        this.user = await UserService.getUserById(this.$route.params.uid);
```
Vergiss nicht Fehler zu behandeln.
denke nen 404 sollte noch gesondert dann behandelt werden.

beanduong replied:
Okay, wird gemacht.

beanduong replied:
Habe eine 404-Komponente erstellt:
![image](https://user-images.githubusercontent.com/8312031/68611036-795fbe00-04b9-11ea-81a5-9076d3deeeb1.png)
Vielleicht können wir anfagen, sie einzubinden.

tbtz commented on src/frontend/components/UserDetail.vue:
```diff
@@ -0,0 +1,59 @@
+          lon: event.longitude,
+          time: event.time
+        }));
+      }
```
👍

tbtz commented on src/frontend/components/UserDetail.vue:
```diff
@@ -0,0 +1,59 @@
+    p {
+      @include textBody;
+      margin: 0 25px 25px 25px;
+      height: 150px;
```
feste größen bei dynamischen inhalten ist meistens nicht sone gute idee
Versuchs mal mit ner min-height. dann könnte es immer noch wachsen falls es mal mehr sein sollte
und falls du nicht explizit wolltest dass bei kurzen descriptions so viel whitespace ist könnte man die min height auch noch veringern. your decision :P
![localhost_3000_user_5dba98d9732d20fbb193eaf9(iPhone X) (2)](https://user-images.githubusercontent.com/9198250/68549832-e2353080-03fc-11ea-89f2-ed91087f2eaa.png)

mustafa89de replied:
vielleicht sollten wir uns allgemein auf die Verwendung relativer Angaben einigen.

tbtz replied:
Was genau meinst du an dieser Stelle mit "rellative Angaben"?

mustafa89de replied:
damit meine ich, dass es vielleicht im Sinne des responsive Designs besser wäre, relative Größenangaben wie %, em etc. zu verwenden statt feste eingabewerte wie 25px.

beanduong replied:
Das mit der fixen Höhe war schon Absicht. Ich damit erreichen, dass die HorizontalEventList immer an gleicher Stelle ist. Ich glaube schon, dass das sehr wichtig ist. Vielleicht können wir die Beschreibung auf eine fixe Anzahl beschränken oder erst mal davon ausgehen, dass die Höhe nicht überschritten wird.

tbtz replied:
overflow: scroll könnte ne lösung sein

beanduong replied:
Grandiose Idee!

mustafa89de commented on src/frontend/pages/UserDetailPage.vue:
```diff
@@ -0,0 +1,55 @@
+    <header>
+      <h1>Profil</h1>
+    </header>
+    <user-detail v-if="user" :user="this.user" />
```
ich glaube das "this" ist bei der Übergabe überflüssig

beanduong replied:
Verzeihung

mustafa89de commented:
Ich denke, Tom hat auf das Wichtigste aufmerksam gemacht. Ansonsten passt es von mir aus 👌

beanduong commented:
Eine Idee für die Optimierung auf Desktop:

![image](https://user-images.githubusercontent.com/8312031/68611688-fa6b8500-04ba-11ea-83f8-28263cd94012.png)

Finde ich gar nicht so schlecht. Man müsste dem Body den roten Hintergrund geben und auf jeder Page die Hintergrundfarbe auf weiß stellen.

beanduong commented:
- removed y-scroll for HorizontalEventList (should work)
- added `NotFound` component (shows 404 and a small description)
- minor visual tweaks on login and registration page
- added 404 component to user detail page when user id was invalid
